x-pack/plugins/actions/server/saved_objects/index.ts

```diff
@@ -98,6 +98,18 @@ export function setupSavedObjects(
   encryptedSavedObjects.registerType({
     type: ACTION_TASK_PARAMS_SAVED_OBJECT_TYPE,
     attributesToEncrypt: new Set(['apiKey']),
+    attributesToIncludeInAAD: new Set([
+      'actionId',
+      'actionExecutionId',
```

Comment:
Not clear to me where the following are coming from, I don't think they are intended to be saved with the SO, not quite clear to me though.

- `actionExecutionId `
- `isEphemeral `
- `taskInfo`
- `actionInfo`

Reply:
Good catch! I think you are correct. Looks like I erroneously got these from the `ExecuteOptions` interface, which is not actually written to the attributes. I see two places where the `ACTION_TASK_PARAMS_SAVED_OBJECT_TYPE` is written:

1. In [`create_execute_function`](https://github.com/elastic/kibana/blob/61fe1307eb80551ba174c6ef25227cfa6d5c48a0/x-pack/plugins/actions/server/create_execute_function.ts#L154)
  actionId: actionToExecute.id,
  params: actionToExecute.params,
  apiKey: actionToExecute.apiKey,
  executionId: actionToExecute.executionId,
  consumer: actionToExecute.consumer,
  relatedSavedObjects: relatedSavedObjectWithRefs,
  ...(actionToExecute.source ? { source: actionToExecute.source.type } : {}),

2. In [`create_unsecured_execute_function`](https://github.com/elastic/kibana/blob/61fe1307eb80551ba174c6ef25227cfa6d5c48a0/x-pack/plugins/actions/server/create_unsecured_execute_function.ts#L131)
  actionId: actionToExecute.id,
  params: actionToExecute.params,
  apiKey: null,
  relatedSavedObjects: relatedSavedObjectWithRefs,
  ...(actionToExecute.source ? { source: actionToExecute.source.type } : {}),

So I will remove the four attributes you mentioned here and reimplement the test. Thanks!

Reply:
Fixed in https://github.com/elastic/kibana/pull/167705/commits/dd8e16ddb8cb1e9aa18db7ce004ea6c89eb2bbe8